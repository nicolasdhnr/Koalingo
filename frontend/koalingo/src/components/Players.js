import React, { useContext } from "react";
import { AuthContext } from "../App";
import styles from '../components/list_styles.css';


export function PlayerTracking(props) {
  
  return (
  <ul>
  {props?.names.map((title) => (
    <li key={title}>
        <h2> {props.reported} </h2>
        <h2>{title}</h2>
      {/* {!description ? null : <p>{description}</p>} */}
    </li>
  ))}
</ul>
  );

}



export default function PlayerList(props) {

  return (
    <ul>
      {props?.players.map((title) => (
        <li key={title}>
            <h2>{title}</h2>
          {/* {!description ? null : <p>{description}</p>} */}
        </li>
      ))}
    </ul>
  );
}
