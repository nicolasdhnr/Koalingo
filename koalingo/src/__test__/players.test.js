import React from 'react';
import {PlayerTracking} from "C:/Users/labas/Documents/GitHub/Koalingo/koalingo/src/components/Players.js";
import { render, cleanup, screen} from '@testing-library/react'
import '@testing-library/jest-dom'


afterEach(cleanup);

it ("player tracking render without crashing", () => {
    const n = ["Saab", "Volvo", "BMW"];
    const r = [0, 0, 0];
    render(<PlayerTracking names = {n} reported = {r} />);
    const play = screen.getByTestId("player");
    expect(play).toBeInTheDocument();
})

it ("Player tracking matches snapshot 1", () => {
    const n = ["Saab", "Volvo", "BMW"];
    const r = [0, 0, 0];
    render(<PlayerTracking names = {n} reported = {r} />);
    const play = screen.getByTestId("player");    
    expect(play).toMatchSnapshot();
})
