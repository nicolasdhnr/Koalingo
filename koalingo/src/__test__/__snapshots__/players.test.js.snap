// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player tracking matches snapshot 1 1`] = `
<ul
  class="player"
  data-testid="player"
>
  <li
    class="playerTrack"
  >
    Saab
    :
     
    0
  </li>
  <li
    class="playerTrack"
  >
    Volvo
    :
     
    0
  </li>
  <li
    class="playerTrack"
  >
    BMW
    :
     
    0
  </li>
</ul>
`;
